eureka2-core/src/main/java/com/netflix/eureka2/channel/AbstractServiceChannel.java

```diff
@@ -60,4 +66,27 @@ public void onNext(T message) {
             }
         });
     }
+
+    protected boolean moveToState(STATE from, STATE to) {
+        if (state.compareAndSet(from, to)) {
+            if (metrics != null) {
+                // We do not track initState (==idle), only subsequent states that
+                // happen when a connection is established.
+                if (from == initState) {
+                    metrics.incrementStateCounter(to);
+                } else {
+                    metrics.stateTransition(from, to);
+                }
+            }
+            return true;
+        }
+        return false;
+    }
+
+    protected boolean moveToState(STATE to) {
+        if (state.get() == to) {
+            return false;
```

Comment:
In this case there would be no point to return the status, as it always would be true. Now we return true only when the state changes. This is useful when we move to Close state, which guards cleanup code. We want to run it only once.